R/scales-.R

```diff
@@ -168,6 +168,39 @@ ScalesList <- ggproto("ScalesList", NULL,
       scale_name <- paste("scale", aes, "continuous", sep = "_")
       self$add(find_global(scale_name, env, mode = "function")())
     }
+  },
+
+  set_palettes = function(self, theme) {
+    for (scale in self$scales) {
+      if (!is.null(scale$palette)) {
+        next
+      }
+
+      # Resolve palette theme setting for this scale
+      type <- if (scale$is_discrete()) "discrete" else "continuous"
+      elem <- paste0("palette.", scale$aesthetics[1], ".", type)
```

Comment:
In the case the scale has 2 aesthetics, say `c("colour", "fill")`, it will only look for `palette.colour.<type>`. Shouldn't it look for the first match instead?

Reply:
Yeah that is a good idea, thanks!